src/NUnitFramework/framework/Internal/ParamAttributeTypeConversions.cs

```diff
@@ -118,8 +118,66 @@ public static bool TryConvert(object? value, Type targetType, [NotNullWhen(true)
                 return convertedValue is not null;
             }
 
+            if (TryGetTupleType(underlyingTargetType, out var tupleTypes))
+            {
+                if (value is object?[] array && array.Length == tupleTypes.Length)
+                {
+                    var success = true;
+                    var ctorArgs = new object?[array.Length];
+                    for (int i = 0; i < ctorArgs.Length; i++)
+                    {
+                        if (!TryConvert(array[i], tupleTypes[i], out ctorArgs[i]))
+                        {
+                            success = false;
+                            break;
+                        }
+                    }
+                    if (success)
+                    {
+                        var ctor = underlyingTargetType.GetConstructor(tupleTypes)!;
+                        convertedValue = ctor.Invoke(ctorArgs);
+                        return true;
+                    }
+                }
+                else if (tupleTypes.Length == 1 && TryConvert(value, tupleTypes[0], out var tupleArg))
+                {
+                    var ctor = underlyingTargetType.GetConstructor(tupleTypes)!;
+                    convertedValue = ctor.Invoke(new[] { tupleArg });
+                    return true;
+                }
+            }
+
             convertedValue = null;
             return false;
         }
+
+        private static bool TryGetTupleType(Type type, [MaybeNullWhen(false)] out Type[] typeArgs)
+        {
+            if (!type.IsGenericType)
+            {
+                typeArgs = null;
+                return false;
+            }
+
+            Type genericType = type.GetGenericTypeDefinition();
+            if (genericType.Assembly == typeof(Tuple).Assembly &&
+                genericType.Namespace == typeof(Tuple).Namespace &&
+                genericType.Name.StartsWith("Tuple`", StringComparison.Ordinal))
+            {
+                typeArgs = type.GetGenericArguments();
+                return true;
+            }
+
+            if (genericType.Assembly == typeof(ValueTuple).Assembly &&
+                genericType.Namespace == typeof(ValueTuple).Namespace &&
+                genericType.Name.StartsWith("ValueTuple`", StringComparison.Ordinal))
+            {
+                typeArgs = type.GetGenericArguments();
+                return true;
+            }
```

Comment:
There should probably be tests of tuples with more than seven elements, since the C# type `(T1, T2, T3, T4, T5, T6, T7, T8)` maps to the type `ValueTuple<T1, T2, T3, T4, T4, T5, T6, T7, ValueTuple<T8>>`.

(System.Tuple and System.ValueTuple require the TRest parameter to be a tuple, and this mapping is recursive to allow any number of elements.)

Reply:
Thanks! Fixed in [14c321b](https://github.com/nunit/nunit/pull/4772/commits/14c321b53f3c91670f19f7a00aa5b1a408a3f36a).